src/platform/test.rs

```diff
@@ -413,6 +413,31 @@ fn receiver_set() {
     }
 }
 
+#[cfg(not(any(feature = "force-inprocess", target_os = "windows", target_os = "android")))]
+fn sigterm_handler(_: libc::c_int) -> () {}
+
+#[test]
+#[cfg(not(any(feature = "force-inprocess", target_os = "windows", target_os = "android")))]
+fn receiver_set_eintr() {
+    let (tx0, rx0) = platform::channel().unwrap();
+    let mut rx_set = OsIpcReceiverSet::new().unwrap();
+    rx_set.add(rx0).unwrap();
+    let parent_pid = unsafe {
+        libc::getpid()
+    };
+    unsafe {
```

Comment:
Not sure it's a good idea to have consecutive `unsafe` blocks... Generally we just use a larger enclosing block, if there are no (non-trivial) safe operations in between.